pkg/notify/elasticsearch.go

```diff
@@ -42,8 +46,36 @@ type ElasticSearch struct {
 
 // NewElasticSearch returns new ElasticSearch object
 func NewElasticSearch(c *config.Config) (Notifier, error) {
+	const AWSService = "es"
+	if c.Communications.ElasticSearch.AWSSigning.Enabled {
+		// Get credentials from environment variables and create the AWS Signature Version 4 signer
+		creds := credentials.NewEnvCredentials()
+		signer := v4.NewSigner(creds)
+		awsClient, err := aws_signing_client.New(signer, nil, AWSService, c.Communications.ElasticSearch.AWSSigning.AWSRegion)
+		if err != nil {
+			return nil, err
+		}
+		elsClient, err := elastic.NewClient(elastic.SetURL(c.Communications.ElasticSearch.Server), elastic.SetScheme("https"), elastic.SetHttpClient(awsClient), elastic.SetSniff(false), elastic.SetHealthcheck(false),elastic.SetGzip(false))
+		if err != nil {
+			return nil, err
+		}
+		return &ElasticSearch{
+			ELSClient:   elsClient,
+			Index:       c.Communications.ElasticSearch.Index.Name,
+			Type:        c.Communications.ElasticSearch.Index.Type,
+			Shards:      c.Communications.ElasticSearch.Index.Shards,
+			Replicas:    c.Communications.ElasticSearch.Index.Replicas,
+			ClusterName: c.Settings.ClusterName,
+		}, nil
```

Comment:
I see duplicate code here. We should just create elsClient here and instead of returning ElasticSearch from here, return at the end of this function (https://github.com/infracloudio/botkube/blob/develop/pkg/notify/elasticsearch.go#L50)